doc/api/events.md

```diff
@@ -1661,12 +1661,23 @@ console.log(listenerCount(myEmitter, 'event'));
 added:
  - v13.6.0
  - v12.16.0
+changes:
+  - version:
+    - v20.0.0
+    pr-url: https://github.com/nodejs/node/pull/41276
+    description: The `close`, `highWatermark`, and `lowWatermark`
+                 options are supported now.
 -->
 
 * `emitter` {EventEmitter}
 * `eventName` {string|symbol} The name of the event being listened for
 * `options` {Object}
   * `signal` {AbortSignal} Can be used to cancel awaiting events.
+  * `close` - {string\[]} Event names that will end the iteration.
+  * `highWatermark` - {integer} **Default:** `Number.MAX_SAFE_INTEGER`
+    The low watermark. The emitter is resumed every time size is lower than it.
```

Comment:
Should this mention that not every event emitter is supported, only streams?

The `highWatermark` option calls `emitter.pause()` and the `lowWatermark` option calls `emitter.resume()`. Those methods are not defined on other event emitters than streams.

Reply:
@ehmicky the watermarks are not on the emitter, it's on the returned async iterator's underlying backing stream.

Reply:
Right, thanks @benjamingr. 

The watermarks represent the number of events currently buffered by the async iterator, and not consumed yet (by using `iterator.next()`). This is distinct from the watermarks set by stream constructors, which represent data currently buffered by the stream, and not read yet (e.g. by using `stream.read()`.

That being said, the `.resume()` and `.pause()` methods are called on the emitter, I believe. So this option requires an event emitter to implement those methods. While a few non-stream event emitters might have those methods (such as `readline.Interface`, for which that option was created), it seems in most cases, the event emitter would need to be a stream.

https://github.com/Farenheith/node/blob/e83baeabb292c8056c14f8477a6271c5c8477649/lib/events.js#L1144

https://github.com/Farenheith/node/blob/e83baeabb292c8056c14f8477a6271c5c8477649/lib/events.js#L1043

I might be misunderstanding this, please let me know.